### PR TITLE
Api based cli update cmd

### PR DESCRIPTION
**Pull request by mike-sul**
Dogfooding the API by implementing the update/finalize CLI commands.
Also, it has helped slightly improve the API.

### CODE REVIEW

**Comment by mike-sul** [comment]
@doanac This lacks of edge cases handling (rollback, boot fw update), but depicts the overall implementation structure and all changes required in the API support it.
Once we agree on this I'll move forward with adding code for edge cases handling.

File: src/cli/cli.cc [hunk 1]
```diff
@@ -1,15 +1,109 @@
...
+  }
+
+  const auto current{client.GetCurrent()};
+  const CheckInResult get_tuf_res{client.CheckIn()};
+  if (get_tuf_res.status == CheckInResult::Status::Failed) {
+    LOG_ERROR << "Failed to pull TUF metadata or they are invalid";
+    return StatusCode::TufMetaPullFailure;
+  }
+
+  // Find TufTarget of the specified version, TODO: consider adding CheckInResult::GetTarget() method
```
Comment (doanac) [inline src/cli/cli.cc:29]: it already exposes a vector of Targets the user can look through. I don't mind introducing friction in an API when the caller is doing something we don't like (not installing the latest).

**Comment by mike-sul** [comment]
@doanac This https://github.com/foundriesio/aktualizr-lite/pull/242/commits/59de13ffc160a835c54bdebf1665622a99f12ab4 is the fix for the `AkliteClient::CONFIG_DIRS` deinit issue.
Effectively, the tests used to link it twice, as part of a api.cc source object and as part of the aklite lib.
This change cleans up our cmake files; build a single library and make use of it by all clients: main, custom client, unit tests.

**Review by doanac — COMMENTED** [review]
seems like the right idea. are you wanting to merge this now - or go ahead and get to the point where you replace the current `update` and `finalize` commands?

**Comment by mike-sul** [comment]
> seems like the right idea. are you wanting to merge this now - or go ahead and get to the point where you replace the current `update` and `finalize` commands?

I opt for the latter.

File: include/aktualizr-lite/api.h [hunk 2]
```diff
@@ -213,6 +233,24 @@ class AkliteClient {
...
+  bool IsInstallationInProgress() const;
+
+  /**
+   * @brief Returns a pending Target if any
+   *
+   * Checks whether there is ongoing installation to be completed and returns corresponding Target.
+   *
+   * @return Pending Target, or "unknown" Target if there is no pending Target
+   */
+  TufTarget GetPendingTarget() const;
```
Comment (doanac) [inline include/aktualizr-lite/api.h:254]: and we don't need any of this in the custom example client because it does the finalization automagically?

**Comment by mike-sul** [comment]
@doanac This is all I wanted to get done in this PR; going to test e2e and on a device.

Also, I can improve the custom client example (daemon) taking into account the API usage experience I've gain while implementing the non-daemon client.

File: src/cli/cli.cc [hunk 3]
```diff
@@ -47,11 +47,11 @@ StatusCode Install(AkliteClient &client, int version) {
   }
 
   TufTarget target;
-  if (version == -1) {
+  if (version == -1 && target_name.empty()) {
     target = cr.GetLatest();
   } else {
     for (const auto &t : cr.Targets()) {
-      if (t.Version() == version) {
+      if (t.Version() == version || t.Name() == target_name) {
```
Comment (doanac) [inline src/cli/cli.cc:54]: I think its fine to continue fighting the version number/name thing like this. Eventually we probably have to build some kind of "version" object that can abstract this.

**Review by doanac — APPROVED** [review]

**Comment by mike-sul** [comment]
@doanac You may wanna look at the last two commits https://github.com/foundriesio/aktualizr-lite/pull/242/commits/eb108dcf3a319df8c0cfa8f324429cffeaea4375 , https://github.com/foundriesio/aktualizr-lite/pull/242/commits/540d3ccaa4aa3c286f216e81bc5ece3f60de15c2

**Review by doanac — APPROVED** [review]

**Pull request merged**